wagtail/models/__init__.py

```diff
@@ -1155,14 +1155,18 @@ def _save_revision(
         previous_revision=None,
         clean=True,
     ):
+        # Need to set this here instead of in _after_save_revision() so that the default
+        # get_admin_display_title() that gets saved as the revision's `title` is correct.
+        self.draft_title = self.title
+
         revision = Revision.objects.create(
             content_object=self,
             base_content_type=get_default_page_content_type(),
             submitted_for_moderation=submitted_for_moderation,
             user=user,
             approved_go_live_at=approved_go_live_at,
             content=self.serializable_data(),
-            title=self.title,
+            title=self.get_admin_display_title(),
```

Comment:
I think the Revision.title should always contain the page's "canonical" title rather than the "admin display" title. It's likely developers will tweak this logic over time and expect those tweaks to take effect immediately. Also, pages created prior to this PR would have their "canonical" title here.